v2/api/dbformysql/v1/user_types.go

```diff
@@ -150,6 +152,104 @@ func (user *User) deleteValidations() []func() error {
 
 // updateValidations validates the update of the resource
 func (user *User) updateValidations() []func(old runtime.Object) error {
+	return []func(old runtime.Object) error{
+		func(old runtime.Object) error {
+			return user.validateIsLocalOrAAD()
+		},
+		user.validateUserTypeNotChanged,
+		user.validateWriteOnePropertiesNotChanged,
+		user.validateUserAADAliasNotChanged,
+	}
+}
+
+// validateWriteOnePropertiesNotChanged function validates the update on WriteOnce properties.
+// TODO: Note this should be kept in sync with admissions.ValidateWriteOnceProperties
+func (user *User) validateWriteOnePropertiesNotChanged(oldObj runtime.Object) error {
+	var errs []error
+
+	oldUser, ok := oldObj.(*User)
+	if !ok {
+		// This shouldn't happen, but if it does don't block things
+		return nil
+	}
+
+	// If we don't have a finalizer yet, it's OK to change things
+	hasFinalizer := controllerutil.ContainsFinalizer(oldUser, genruntime.ReconcilerFinalizer)
+	if !hasFinalizer {
+		return nil
+	}
+
+	if oldUser.Spec.AzureName != user.Spec.AzureName {
+		errs = append(errs, errors.Errorf("updating 'AzureName' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldUser.GetName()))
+	}
+
+	// Ensure that owner has not been changed
+	oldOwner := oldUser.Owner()
+	newOwner := user.Owner()
+
+	bothHaveOwner := oldOwner != nil && newOwner != nil
+	ownerAdded := oldOwner == nil && newOwner != nil
+	ownerRemoved := oldOwner != nil && newOwner == nil
+
+	if (bothHaveOwner && oldOwner.Name != newOwner.Name) || ownerAdded {
+		errs = append(errs, errors.Errorf("updating 'Owner.Name' is not allowed for '%s : %s", oldObj.GetObjectKind().GroupVersionKind(), oldUser.GetName()))
```

Comment:
For illustration, suggesting the same change here:

```suggestion
		err := errors.Errorf(
			"updating 'Owner.Name' is not allowed for '%s : %s", 
			oldObj.GetObjectKind().GroupVersionKind(), 
			oldUser.GetName())
		errs = append(errs, err)
```

Reply:
Done